generator.yaml

```
@@ -96,6 +97,9 @@ resources:
       errors:
         404:
           code: NoSuchBucket
+      terminal_codes:
+        - PermanentRedirect
+        - InvalidLocationConstraint
```

can we also add the error from the github issue as terminal?
```suggestion
        - InvalidLocationConstraint
        - MalformedXML
```

---

Not sure if `PermanentRedirect` is a terminal error

---

without `PermanentRedirect`, Bucket with `services.k8s.aws/region: us-east-1` and `locationConstraint: us-west-1` keeps reconciling. 

> The bucket you are attempting to access must be addressed using the specified endpoint.